OSDOCS-19870: adds main console docs RHCL (#112113)
openshift/openshift-docs: modules/ref-rhcl-api-management-auth-methods.adoc

//
// * develop/rhcl-api-management.adoc

:_mod-docs-content-type: REFERENCE
[id="ref-rhcl-api-management-auth-methods_{context}"]
= API management authentication methods

[role="_abstract"]
You can use one of two authentication methods for protecting APIs with the {prodname} {ocp} web console plugin. Configure your chosen method at the platform level through `AuthPolicy` custom resources (CRs) that are automatically discovered by the {prodname} controller.

[id="rhcl-api-key-authentication_{context}"]
== API key authentication

API key authentication uses Kubernetes secrets to store credentials. This method is suitable for internal APIs, development environments, or scenarios that require fine-grained control over API access.

//Q: Can we see and interact with all steps in the console?
The following list is an example workflow:

. An API consumer creates an `APIKey` resource requesting access to an `APIProduct` resource.
. Depending on the `APIProduct` resource `approvalMode` field one of the following events happens:

* *Automatic*: The controller immediately approves the request.
* *Manual*: The request enters `Pending` state, awaiting API owner approval.

> . For manual approval, the API owner or administrator creates an `APIKeyApproval` resource to approve or reject the request.

Reviewer: While this is technically correct, from the console user perspective, it's just selecting a pending api key request and pushing a button that says: "approve".

// Module included in the following assemblies:
//
// * develop/rhcl-api-management.adoc

:_mod-docs-content-type: REFERENCE
[id="ref-rhcl-api-management-auth-methods_{context}"]
= API management authentication methods

[role="_abstract"]

> You can use one of two authentication methods for protecting APIs with the {prodname} {ocp} web console plugin. Configure your chosen method at the platform level through `AuthPolicy` custom resources (CRs) that are automatically discovered by the {prodname} controller.

Reviewer 2: An attempt to avoid passive voice:
```suggestion
You can use one of two authentication methods for protecting APIs with the {prodname} {ocp} web console plugin. Configure your chosen method at the platform level through `AuthPolicy` custom resources (CRs), which the {prodname} controller automatically discovers.
```

[id="rhcl-api-key-authentication_{context}"]
== API key authentication

API key authentication uses Kubernetes secrets to store credentials. This method is suitable for internal APIs, development environments, or scenarios that require fine-grained control over API access.

The following list is an example workflow:

. An API consumer creates an `APIKey` resource requesting access to an `APIProduct` resource.

> . Depending on the `APIProduct` resource `approvalMode` field one of the following events happens:

Reviewer 2: ```suggestion
. Depending on the `APIProduct` resource `approvalMode` field, one of the following events happens:
```

* *Automatic*: The controller immediately approves the request.
* *Manual*: The request enters `Pending` state, awaiting API owner approval.

. For manual approval, the API owner or administrator clicks *Approve* in the web console.
. After approval, credentials become available to the API consumer.
. The API consumer retrieves the key from the console and uses it in API requests.
. The `AuthPolicy` CR validates incoming requests against the credentials.

[id="rhcl-oidc-jwt-authentication_{context}"]
== OIDC and JWT authentication

OpenID Connect (OIDC) authentication delegates credential management to an external identity provider. This method is suitable for APIs that integrate with existing identity providers, require stronger authentication, or need integration with enterprise single sign-on (SSO) systems. No `APIKey` resources are created. Access control happens at the identity provider level.

There is no request or approval workflow in the {ocp} web console for this authentication method, but certain parts of the authentication process are visible. The following list is an example workflow:

> . A cluster administrator configures an `AuthPolicy` resource with JWT validation pointing to an OIDC issuer.

Reviewer 2: Do you need to spell out JWT, or is that terminology that most of your audience will be familiar with?

. The {prodname} controller discovers the JWT authentication scheme and performs OIDC discovery to find the token endpoint.
. Discovered authentication details are surfaced in the `APIProduct` resource status.
. An API consumer views the identity provider URL and token endpoint in the web console.
. The API consumer obtains a JWT token directly from the identity provider. For example, by using client credentials flow or any other available flow.
. The consumer uses the JWT token in API requests.
. The `AuthPolicy` CR validates the token signature and claims against the OIDC issuer.

[IMPORTANT]
====

> You can create an `OIDCPolicy` object in the web console, however this policy is experimental and not supported. If you want to use OIDC authentication in your {prodname} deployment, using it with an `AuthPolicy` is the optimized path.

Reviewer 2: ```suggestion
You can create an `OIDCPolicy` object in the web console, but this policy is experimental and not supported. If you want to use OIDC authentication in your {prodname} deployment, use it with an `AuthPolicy` object.
```
Feel free to ignore if my revision is inaccurate.
